libmamba/src/core/pool.cpp

```diff
@@ -160,6 +160,31 @@ namespace mamba
             return false;
         }
 
+        bool subdir_match(std::string candidate_repo_url, std::string needle_spec)
+        {
+            // Example candidate_repo_url: https://.../conda-forge/linux-64
+            // example needle_spec: conda-forge/osx-64::xtensor
+
+            std::string needle_channel = split(needle_spec, ":", 1)[0];
+            if (!contains(needle_channel, "/"))
+            {
+                // Subdir not specified, so any subdir is fine
+                return true;
+            }
+            std::string needle_subdir = rsplit(needle_channel, "/", 1)[1];
```

Comment:
@jonashaag, concerning https://github.com/mamba-org/mamba/issues/2431,  I think this is the problematic part. It assumes that if the channel name contains a slash it's because it is specifying a subdir, but `pkgs/main` is a valid channel name (whether we like it or not 😬). So I'd say that we add some logic after this to check whether `needle_subdir` is indeed a known subdir, and ignore it otherwise.

Reply:
The logic implemented in `conda` can be found at https://github.com/conda/conda/blob/a9b1838f059501528e6f48c35972fd5f379dec0a/conda/models/match_spec.py#L188-L199

Reply:
Sounds correct. That's really unfortunate :/

Reply:
Are there any known channels with `/` other than the official Anaconda ones? I wonder if we could get away with hardcoding them if the subdir lookup thing is too complicated to implement.

Reply:
@jaimergp I did some investigation. This part of the code is never run, I think because this PR collided with some other refactorings that were done. So while incorrect, it's not the cause of the bug.

Reply:
TIL `KNOWN_PLATFORMS` in `channel.cpp`